src/cdk/scrolling/virtual-scroll-viewport.ts

```diff
@@ -202,6 +204,17 @@ export class CdkVirtualScrollViewport extends CdkVirtualScrollable implements On
       this.elementRef.nativeElement.classList.add('cdk-virtual-scrollable');
       this.scrollable = this;
     }
+
+    const ref = effect(
+      () => {
+        if (!this._changeDetectionNeeded()) {
+          return;
+        }
+        this._doChangeDetection();
+      },
+      {injector: inject(ApplicationRef).injector},
+    );
+    inject(DestroyRef).onDestroy(() => void ref.destroy());
```

Comment:
nit: should be remove the `void`?

Reply:
This one's kind of the internal tslint style where there's technically a risk of `onDestroy` paying attention to the return value and `destroy` being refactored to return a value that would affect the outcome. I can change this to `() => { ... }` instead if you want.

Reply:
Yep that's fine, we just don't do that anywhere else.

Reply:
Yea 😅 it's just a habit I picked up so I don't get the linter complaining to me all the time